palette/src/js/script.js

```diff
@@ -0,0 +1,409 @@
+class Palette {
+  constructor() {
+    this.mode = 'pensil';
+    this.currentColor = '#000000';
+    this.prevColor = '#ffffff';
+    this.permitDraw = false;
+    this.canvas = null;
+    this.ctx = null;
+    this.scale = 32;
+    this.firstPoint = {
+      left: null,
+      top: null,
+    };
+    this.busy = false;
+  }
+
+  // body load an other events
+  bodyLoad() {
+    this.createCanvas();
+    if (localStorage.getItem('state')) {
+      const state = JSON.parse(localStorage.getItem('state'));
+      this.mode = state.mode;
+      this.currentColor = state.currentColor;
+      this.prevColor = state.prevColor;
+      this.ctx.fillStyle = this.currentColor;
+      document.getElementById('inp-color').value = this.currentColor;
+    }
+    this.toggleTool(document.querySelector(`#${this.mode}`));
+    document.querySelector('#current-color i').style.color = this.currentColor;
+    document.querySelector('#prev-color i').style.color = this.prevColor;
+    document.querySelector('.left-menu').addEventListener('mousedown', (event) => { this.toggleTool(event); });
+    document.querySelector('.left-bottom-menu').addEventListener('mousedown', (event) => { this.toggleColor(event); });
+    document.getElementById('inp-color').addEventListener('change', (event) => { this.inputColor(event); });
```

Comment:
The best option is to query elements once and store them in the class as I've noticed that you query them once again